assets/src/data/eventespresso/base-entities.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { upperFirst, camelCase, reduce, isMap, isNaN } from 'lodash';
+import pluralize from 'pluralize';
+import { mapReducer } from '@eventespresso/eejs';
+
+/**
+ * A helper for getting a method name.
+ * @param {string} modelName
+ * @param {string} suffix
+ * @param {string} prefix
+ * @param {boolean} usePlural
+ * @return {string} Returns a name for a method.
+ */
+export const getMethodName = (
+	modelName,
+	suffix = '',
+	prefix = 'get',
+	usePlural = false
+) => {
+	modelName = usePlural ? pluralize( modelName ) : modelName;
+	return prefix + upperFirst( camelCase( modelName + upperFirst( suffix ) ) );
+};
+
+/**
+ * Given a collection of existing entities and a collection of incoming
+ * entities, this returns a merged object with preference given to common
+ * entities from the existingEntities collection.
+ *
+ * Incoming collections can be Maps or plain objects.
+ *
+ * Note if incomingEntities is a Map, the ORDER of the map will be preserved
+ * even if the values of entities in the map are replaced by values from
+ * existing entities.
+ *
+ * @param {Map|Object} existingEntities
+ * @param {Map|Object} incomingEntities
+ * @return {Map} A new collection of entities. Note if existing entities came in
+ * as a plain object, this returns a Map.
+ */
+export const keepExistingEntitiesInObject = (
+	existingEntities,
+	incomingEntities,
+) => {
+	const hasThenGetEntity = ( defaultEntity, entityId ) => {
```

Comment:
LOL, I seriously almost fell out of my chair when I read this.  Ya I was pretty mind warped when I wrote this.  The idea behind the name was "if the existing entities collection **has** an entity matching this id, then **get** it, otherwise use default".  So yea your suggested name is better, I'll rename.